include/openthread/message.h

```diff
@@ -109,6 +109,18 @@ typedef struct otMessageSettings
     otMessagePriority mPriority;            ///< The message priority level.
 } otMessageSettings;
 
+/**
+ * This type defines the callback for a message. The callback is invoked to indicate the status of an outbound message,
```

Comment:
nit: brief overview should be on its own line, followed by empty line, followed by more detailed description